packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -170,7 +174,52 @@ export default class Editor {
      * @returns true if node is inserted. Otherwise false
      */
     public insertNode(node: Node, option?: InsertOption): boolean {
-        return node ? this.core.api.insertNode(this.core, node, option) : false;
+        let darkModeNormalize = null;
+        if (this.isDarkMode()) {
+            darkModeNormalize = this.convertContentToDarkMode(node);
+        }
+
+        const result = node ? this.core.api.insertNode(this.core, node, option) : false;
+
+        if (result && darkModeNormalize) {
+            darkModeNormalize();

```

Comment:
curious why you need to bind the darkModeNormalize node argument before insert.
Can you explain why + document with a comment?

Reply:
And if you don't you could simplify `convertToDarkMode` by moving the ParentNode type check outside of it.

Reply:
The reason I do it this way is a Node can be one of three things (at least, 3 useful things that we have to do this transform on)

1) An Element (usually an HTML Element)
2) A Document (unlikely, but possible)
3) A DocumentFragment (we do this sometimes)

The first 2, I can get live collections of and I can reliably guarantee that they will still exist even after insertions.

But, when a DocumentFragment is inserted into a document using DOM manipulation APIs, it becomes empty. It's a special cased sort of node that has its _contents_ put in to the document instead of the node itself.

This means that you can't get any of the nodes that exist on it after it's been inserted. You have to get a live list, and then clone the list into an array, and then you can manipulate it afterwords.

I agree it's not clear and I was meaning to put in a comment, but here's the context. If there's a good way to get around it, I'm open to it.